zaza/openstack/charm_tests/magnum/tests.py

```diff
@@ -0,0 +1,175 @@
+        # Verify new image name
+        images_list = list(self.glance_client.images.list())
+        self.assertEqual(images_list[0].name, IMAGE_NAME,
+                         "glance image create failed or unexpected")
+
+        # Create magnum template
+        template_fields = {
+             'name': TEMPLATE_NAME,
+             'image_id': IMAGE_NAME,
+             'external_network_id': 'ext_net',
+             'dns_nameserver': '1.1.1.1',
+             'master_flavor_id': FLAVOR_NAME,
+             'flavor_id': FLAVOR_NAME,
+             'coe': 'kubernetes'
+         }
+
+        logging.info('Creating magnum cluster template...')
+
+        template = self.magnum_client.cluster_templates.create(**template_fields)
+        logging.info('Cluster template data: {}'.format(template))
+        template_id = template.uuid
+
+        # Create a magnum cluster from a magnum template, verify its status
+        logging.info('Creating magnum cluster...')
+           # Create the cluster
+        cluster_fields = {
+            'name': CLUSTER_NAME,
+            'cluster_template_id': template_id,
+            'master_count': 1,
+            'node_count': 1,
+            'keypair': 'zaza'
+        }
+
+        cluster = self.magnum_client.clusters.create(**cluster_fields)
+        logging.info('Cluster data: {}'.format(cluster))
+        cluster_id = cluster.uuid
+
+        # Confirm stack reaches COMPLETE status.
+        openstack_utils.resource_reaches_status(self.magnum_client.clusters,
+                                                cluster_id,
+                                                expected_status="CREATE_COMPLETE",
+                                                msg="Cluster status wait",
+                                                stop_after_attempt=20,
+                                                wait_iteration_max_time=600,
+                                                wait_exponential_multiplier=2)
+
+        # List cluster
+        clusters = list(self.magnum_client.clusters.list())
+        logging.info('All clusters: {}'.format(clusters))
+
+        # Get cluster information
+        cluster = self.magnum_client.clusters.get(CLUSTER_NAME)
+
+        # Confirm cluster name.
+        logging.info('Expected, actual cluster name: {}, '
+                     '{}'.format(CLUSTER_NAME, cluster.name))
+        self.assertEqual(cluster.name, CLUSTER_NAME,
+                         'Cluster name mismatch, '
+                         '{} != {}'.format(CLUSTER_NAME, cluster.name))
+
+        # Confirm magnum cluster stats
+        logging.info('Confirming magnum cluster stats...')
+        stats = self.magnum_client.stats.list().to_dict()
+        expected_stats =  {'clusters': 1, 'nodes': 2}
+        self.assertEqual(expected_stats, stats,
+                         'Cluster stats mismatch, '
+                         '{} != {}'.format(expected_stats, stats))
+
```

Comment:
Add another assertion here to get the cluster `kubeconfig`, and check if the Kubernetes API port is reachable.

You need to parse the `kubeconfig` file (which is YAML format) for the Kubernestes API host / port.